libsolidity/ast/Types.cpp

```diff
@@ -2607,6 +2607,41 @@ bool FunctionType::isExplicitlyConvertibleTo(Type const& _convertTo) const
 	return _convertTo.category() == category();
 }
 
+bool FunctionType::isImplicitlyConvertibleTo(Type const& _convertTo) const
+{
+	if (_convertTo.category() != category())
+		return false;
+	
+	FunctionType const& convertTo = dynamic_cast<FunctionType const&>(_convertTo);
+	
+	if (
+		m_kind != convertTo.m_kind ||
+		m_parameterTypes.size() != convertTo.m_parameterTypes.size() ||
+		m_returnParameterTypes.size() != convertTo.m_returnParameterTypes.size()
+	)
+		return false;
+	
+	auto typeCompare = [](TypePointer const& _a, TypePointer const& _b) -> bool { return *_a == *_b; };
+	if (
+		!equal(m_parameterTypes.cbegin(), m_parameterTypes.cend(), convertTo.m_parameterTypes.cbegin(), typeCompare) ||
+		!equal(m_returnParameterTypes.cbegin(), m_returnParameterTypes.cend(), convertTo.m_returnParameterTypes.cbegin(), typeCompare)
+	)
+		return false;
+	
+	if (m_gasSet != convertTo.m_gasSet || m_valueSet != convertTo.m_valueSet)
+		return false;
+	if (bound() != convertTo.bound())
+		return false;
+	if (bound() && *selfType() != *convertTo.selfType())
```

Comment:
The check for `selfType` should actually be included in the parameterTypes check, but we should at least have an assertion here about that.

Reply:
I've turned it into an assertion